config_types.h

```diff
@@ -93,4 +93,20 @@ struct server_addr {
     int m_last_error;
 };
 
+struct config_cpu_list {
+    std::vector<unsigned int> cpu_list;
+    std::vector<unsigned int>::iterator next_cpu;
+
+    config_cpu_list() {}
+    explicit config_cpu_list(unsigned int cores);
+    explicit config_cpu_list(std::string str);
```

Comment:
for those who are not familiar with all c++ features, can you please explain what is the purpose of the `explicit` specifier?

Reply:
explicit is used in-order to prevent the constructor from being used implicitly for automatic conversions